src/scheduler/server_info.c

```diff
@@ -704,21 +712,91 @@ query_server_dyn_res(server_info *sinfo)
 			/* Make sure file does not have open permissions */
 			err = tmp_file_sec(filename, 0, 1, S_IWGRP|S_IWOTH, 1);
 			if (err != 0) {
-				log_eventf(PBSEVENT_SECURITY, PBS_EVENTCLASS_SERVER, LOG_ERR, "server_dyn_res", 
+				log_eventf(PBSEVENT_SECURITY, PBS_EVENTCLASS_SERVER, LOG_ERR, "server_dyn_res",
 					"error: %s file has a non-secure file access, setting resource %s to 0, errno: %d",
 					filename, res->name, err);
-				(void) set_resource(res, res_zero, RF_AVAIL);
+				set_resource(res, res_zero, RF_AVAIL);
+				continue;
+			}
+
+			if (pipe(pdes) < 0) {
+				pipe_err = errno;
+				goto end_of_exec;
+			}
+			switch(pid = fork()) {
+			case -1:	/* error */
+				close(pdes[0]);
+				close(pdes[1]);
+				pipe_err = errno;
+				goto end_of_exec;
+				break;
+			case 0:		/* child */
+				close(pdes[0]);
+				if (pdes[1] != STDOUT_FILENO) {
+					dup2(pdes[1], STDOUT_FILENO);
+					close(pdes[1]);
+				}
+				setpgid(0, 0);
+				if (sigemptyset(&allsigs) == -1) {
+					perror("sigemptyset");
+					exit(1);
+				}
+				if (sigprocmask(SIG_SETMASK, &allsigs, NULL) == -1) {	/* unblock all signals */
+					perror("sigprocmask");
+					exit(1);
+				}
+
+				char *argv[4];
+				argv[0] = "/bin/sh";
+				argv[1] = "-c";
+				argv[2] = conf.dynamic_res[i].command_line;
+				argv[3] = NULL;
+
+				execve("/bin/sh", argv, environ);
+				_exit(127);
+			}
+
+			FD_ZERO(&set);
+			FD_SET(pdes[0], &set);
+			log_eventf(PBSEVENT_ERROR, PBS_EVENTCLASS_SERVER, LOG_DEBUG, "server_dyn_res",
+				"timeout is %d", server_dyn_res_alarm);
+			if (server_dyn_res_alarm) {
+				struct timeval timeout;
+				timeout.tv_sec = server_dyn_res_alarm;
+				timeout.tv_usec = 0;
+				ret = select(FD_SETSIZE, &set, NULL, NULL, &timeout);
+			} else {
+				ret = select(FD_SETSIZE, &set, NULL, NULL, NULL);
+			}
+			if (ret == -1) {
+				log_eventf(PBSEVENT_ERROR, PBS_EVENTCLASS_SERVER, LOG_DEBUG, "server_dyn_res",
+				"Select() failed for script %s", conf.dynamic_res[i].command_line);
+			} else if (ret == 0) {
+				log_eventf(PBSEVENT_DEBUG, PBS_EVENTCLASS_SERVER, LOG_DEBUG, "server_dyn_res",
+				"Program %s timed out", conf.dynamic_res[i].command_line);
```

Comment:
Say that you are setting the resource to zero.

Reply:
It says it gets set to zero on line 820.

Reply:
Actually, it even gets set to zero down there, so I don't need to set it up here.